### PR TITLE
doc: add manifests and improve readme

### PR DESCRIPTION
**Pull request by huanggze**
Signed-off-by: huanggze <loganhuang@yunify.com>

### Doc Updates
- Add manifests folder for quick start
- Add missing links in API docs
- Refine README
- Add missing `COPY pkg/ pkg/` in Dockerfile

### Code Changes
- Update `cmd/doc-gen/main.go`
- Mount position db file only if `fluentbit.spec.positionDB` is specified
- Add service section to the final config only if `fluentbitconfig.spec.service` is specified
-  Add missing kubebuilder comments for properly generating clusterroles/clusterrolebindings
- To avoid potential conflicts, name clusterroles/clusterrolebindings as `kubesphere:operator:fluentbit-operator`, rather than `fluentbit-operator`. The same with Fluent Bit.

Seems that authorizing is a big problem. There is no need to grant exensive roles if the purpose for deploying Fluent Bit is not to collect container logs.

### CODE REVIEW

**Comment by benjaminhuo** [comment]
Great

File: manifests/logging-stack/filter-kubernetes.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      labels: false
+      annotations: false
+  - nest:
+      operation: lift
+      nestedUnder: kubernetes
+      addPrefix: kubernetes_
+  - modify:
+      rules:
+      - remove: stream
+      - remove: kubernetes_pod_id
```
Comment (benjaminhuo) [inline manifests/logging-stack/filter-kubernetes.yaml:25]: duplicated pod id？

File: manifests/logging-stack/output-kafka.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+metadata:
+  name: kafka
+  namespace: kubesphere-logging-system
+  labels:
+    logging.kubesphere.io/enabled: "false"
+    app.kubernetes.io/version: v0.2.0
+spec:
+  match: kube.*
+  kafka:
+    brokers: 192.168.1.3:9092, 192.168.1.4:9092
```
Comment (benjaminhuo) [inline manifests/logging-stack/output-kafka.yaml:12]: there is an space after ',' 
it's not necessary?

File: README.md [hunk 3]
```diff
@@ -2,27 +2,170 @@
...
+
+|Dir Path|Description|
+|---|---|
+|/fluent-bit/tail|Stores tail related files, eg. file tracking db. Using [fluentbit.spec.positionDB](docs/crd.md#fluentbitspec) will mount a file `pos.db` under this dir by default.|
+|/fluent-bit/secrets/{secret_name}|Stores secrets, eg. TLS files. Specify secrets to mount in [fluentbit.spec.secrets](docs/crd.md#fluentbitspec), then you have access.|
+|/fluent-bit/config|Stores the final config file.|
+
+> Note that ServiceAccount files are mounted at `/var/run/secrets/kubernetes.io/serviceaccount`.
+
+## Features In Plan
```
Comment (benjaminhuo) [inline README.md:144]: 3 aws fluentbit plugin should be in the plan:
amazon-kinesis-firehose-for-fluent-bit: https://github.com/aws/amazon-kinesis-firehose-for-fluent-bit
amazon-cloudwatch-logs-for-fluent-bit: https://github.com/aws/amazon-cloudwatch-logs-for-fluent-bit
amazon-kinesis-streams-for-fluent-bit: https://github.com/aws/amazon-kinesis-streams-for-fluent-bit

these plugins was provided in amazon/aws-for-fluent-bit container image: https://github.com/aws/aws-for-fluent-bit, we could investigate if it's possible to integrate these 3 plugins without using aws fluentbit image

File: README.md [hunk 4]
```diff
@@ -2,27 +2,170 @@
...
+### Quick Start
+
+The quick start instructs you to deploy fluent bit with dummy as input and stdout as output, which is equivalent to execute the binary with `fluent-bit -i dummy -o stdout`. 
+
+```shell
+kubectl apply -f manifests/setup
+kubectl apply -f manifests/quick-start
+```
+
+Once everything up, inspect logs of fluent bit pods. You should observe loads of messages like below:
```
[the remark this answers is not in the document]
Comment (huanggze) [reply README.md]: done

File: README.md [hunk 5]
```diff
@@ -2,27 +2,170 @@
...
+
+Success!
+
+### Logging Stack
+
+This guide provisions logging pipeline for your work environment. It installs Fluent Bit as daemonset for collecting container logs, filtering unneeded fields and forwarding them to the target destinations (eg. es, kafka, fluentd).
+
+![logging stack](docs/images/logging-stack.svg)
+
+Note that you need a running elasticsearch v5+ to receive data before start. **Remember to adapt [output-elasticsearch.yaml](manifests/logging-stack/output-elasticsearch.yaml) to your es setup**. Otherwise fluent bit will spam errors. Kafka and Fluentd are optional and switched off by default.
```
[the remark this answers is not in the document]
Comment (huanggze) [reply README.md]: done

File: manifests/logging-stack/filter-kubernetes.yaml (hunk 1 above)
Comment (huanggze) [reply manifests/logging-stack/filter-kubernetes.yaml:25]: done

File: manifests/logging-stack/output-kafka.yaml (hunk 2 above)
Comment (huanggze) [reply manifests/logging-stack/output-kafka.yaml]: done

File: README.md (hunk 3 above)
Comment (huanggze) [reply README.md:144]: Update readmap: Support custom Input/Filter/Output plugins

**Pull request merged**